tests/unit/test_bugjira.py

```diff
@@ -180,6 +182,94 @@ def test_get_issue_with_non_string_key(sandboxed_bugjira):
         sandboxed_bugjira.get_issue(1)
 
 
+def test_add_comment_non_issue(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a None value for the 'issue' parameter
+    THEN a ValueError should be raised
+    """
+    with pytest.raises(ValueError,
+                       match="issue must be an Issue"):
+        sandboxed_bugjira.add_comment(None, "")
+
+
+def test_add_comment_non_string_comment(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a None value for the 'comment' parameter
+    THEN a ValueError should be raised
+    """
+    with pytest.raises(ValueError,
+                       match="comment must be a str"):
+        sandboxed_bugjira.add_comment(Issue(key="foo"),
+                                      None)
+
+
+def test_add_comment_good_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we call add_comment with a BugzillaIssue and comment string
+    THEN the bugzilla backend's build_update and update_bugs methods should
+        each be invoked once
+    """
+    comment_text = "This is a sample comment"
+    key = "123456"
+    issue = BugzillaIssue(key=key)
+    sandboxed_bugjira.add_comment(issue, comment_text)
+    assert sandboxed_bugjira.bugzilla.build_update.call_count == 1
+    assert sandboxed_bugjira.bugzilla.update_bugs.call_count == 1
+
+
+def test_add_comment_bugzilla_exception(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance whose bugzilla backend's update_bugs method
+        raises an Exception
+    WHEN we call add_comment
+    THEN a BrokerAddCommentException should be raised
+    """
+    def raise_exception(*args):
+        raise Exception("Exception raised")
```

Comment:
You don't need to do this here, you should be able to leverage '[side_effect](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect)' on the next line:

    sandboxed_bugjira.bugzilla.update_bugs.side_effect = Exception('Exception raised')